packages/sirius-web/backend/sirius-web-application/src/main/java/org/eclipse/sirius/web/application/editingcontext/services/EventProcessorSubscriptionProvider.java

```diff
@@ -0,0 +1,101 @@
+import org.eclipse.sirius.web.application.capability.services.CapabilityVote;
+import org.eclipse.sirius.web.application.capability.services.api.ICapabilityVoter;
+import org.eclipse.sirius.web.application.library.services.api.ILibraryEditingContextService;
+import org.eclipse.sirius.web.application.project.services.api.IProjectEditingContextService;
+import org.eclipse.sirius.web.domain.services.api.IMessageService;
+import org.springframework.stereotype.Service;
+
+import reactor.core.publisher.Flux;
+import reactor.core.scheduler.Schedulers;
+
+/**
+ * Used to provide the subscription of a representation in an editing context.
+ *
+ * @author gcoutable
+ */
+@Service
+public class EventProcessorSubscriptionProvider implements IEventProcessorSubscriptionProvider {
+
+    private final IEditingContextEventProcessorRegistry editingContextEventProcessorRegistry;
+
+    private final IProjectEditingContextService projectEditingContextService;
+
+    private final ILibraryEditingContextService libraryEditingContextService;
+
+    private final List<ICapabilityVoter> capabilityVoters;
+
+    private final List<IRepresentationEventProcessorFluxCustomizer> representationEventProcessorFluxCustomizers;
+
+    private final IMessageService messageService;
+
+    public EventProcessorSubscriptionProvider(IEditingContextEventProcessorRegistry editingContextEventProcessorRegistry, IProjectEditingContextService projectEditingContextService,
+            ILibraryEditingContextService libraryEditingContextService, List<ICapabilityVoter> capabilityVoters,
+            List<IRepresentationEventProcessorFluxCustomizer> representationEventProcessorFluxCustomizers, IMessageService messageService) {
+        this.editingContextEventProcessorRegistry = Objects.requireNonNull(editingContextEventProcessorRegistry);
+        this.projectEditingContextService = Objects.requireNonNull(projectEditingContextService);
+        this.libraryEditingContextService = Objects.requireNonNull(libraryEditingContextService);
+        this.capabilityVoters = Objects.requireNonNull(capabilityVoters);
+        this.representationEventProcessorFluxCustomizers = Objects.requireNonNull(representationEventProcessorFluxCustomizers);
+        this.messageService = Objects.requireNonNull(messageService);
+    }
+
+    @Override
+    public Flux<IPayload> getSubscription(String editingContextId, String representationId, IInput input) {
+
+        var canView = false;
+        var optionalProjectId = this.projectEditingContextService.getProjectId(editingContextId);
+        if (optionalProjectId.isPresent()) {
+            canView = this.capabilityVoters.stream().allMatch(voter -> voter.vote(SiriusWebCapabilities.PROJECT, optionalProjectId.get(), SiriusWebCapabilities.Project.VIEW) == CapabilityVote.GRANTED);
+        } else {
+            var optionalLibraryId = this.libraryEditingContextService.getLibraryId(editingContextId);
+            if (optionalLibraryId.isPresent()) {
```

Comment:
The check should not be performed with this id but instead with `namespace:name:version`

Reply:
Done